apps/admin-gui/src/app/admin/pages/admin-page/admin-searcher/admin-searcher.component.ts

```diff
@@ -0,0 +1,164 @@
+import { Component, OnInit } from '@angular/core';
+import {
+  AttributeDefinition,
+  AttributesManagerService,
+  EnrichedFacility,
+  RichMember,
+  RichUser,
+  SearcherService,
+  Vo,
+  VosManagerService,
+} from '@perun-web-apps/perun/openapi';
+import { compareFnDisplayName, compareFnName } from '@perun-web-apps/perun/utils';
+import {
+  TABLE_SEARCHER_FACILITIES,
+  TABLE_SEARCHER_MEMBERS,
+  TABLE_SEARCHER_RESOURCES,
+  TABLE_SEARCHER_USERS,
+} from '@perun-web-apps/config/table-config';
+import { ResourceWithStatus } from '@perun-web-apps/perun/models';
+
+@Component({
+  selector: 'app-admin-searcher',
+  templateUrl: './admin-searcher.component.html',
+  styleUrls: ['./admin-searcher.component.scss'],
+})
+export class AdminSearcherComponent implements OnInit {
+  constructor(
+    private attributesManager: AttributesManagerService,
+    private searcher: SearcherService,
+    private voService: VosManagerService
+  ) {}
+
+  loading: boolean;
+  loadingEntityData = false;
+  allAttrDefinitions: AttributeDefinition[] = [];
+  availableAttrDefs: AttributeDefinition[] = [];
+  searchInput: { [p: string]: string };
+  tableId = TABLE_SEARCHER_USERS;
+  entities: RichUser[] | RichMember[] | EnrichedFacility[] | ResourceWithStatus[] = [];
+  vos: Vo[] = [];
+  selectedVo: Vo;
+
+  ngOnInit(): void {
+    this.loading = true;
+    this.attributesManager.getAllAttributeDefinitions().subscribe((attrDefs) => {
+      this.allAttrDefinitions = attrDefs;
+      this.filterAttrDefinitions('user');
+      this.loading = false;
+    });
+  }
+
+  filterAttrDefinitions(entity: string): void {
+    this.availableAttrDefs = this.allAttrDefinitions
+      .filter((attrDef) => attrDef.entity === entity)
+      .sort(compareFnDisplayName);
+  }
+
+  tabChanged(indexChangeEvent: number): void {
+    this.entities = [];
+    switch (indexChangeEvent) {
+      case 0: // Users
+        this.filterAttrDefinitions('user');
+        this.tableId = TABLE_SEARCHER_USERS;
+        break;
+      case 1: // Members
+        this.loadingEntityData = true;
+        this.filterAttrDefinitions('user'); // we search members in searcher by user attributes !!!
+        this.loadAllVos();
+        this.tableId = TABLE_SEARCHER_MEMBERS;
+        break;
+      case 2:
+        this.filterAttrDefinitions('facility');
+        this.tableId = TABLE_SEARCHER_FACILITIES;
+        break;
+      case 3:
+        this.filterAttrDefinitions('resource');
+        this.tableId = TABLE_SEARCHER_RESOURCES;
+        break;
+      default:
+        break;
+    }
+  }
+
+  loadAllVos(): void {
+    this.loadingEntityData = true;
+    this.voService.getAllVos().subscribe((vos) => {
+      this.vos = vos.sort(compareFnName);
```

Comment:
question to think about - don't we want to be the organizations sorted like that by default in the vo-search-select-component ?

Reply:
Good point. FYI in the `notifications-copy-mails-dialog` and `application-form-copy-items-dialog` components, some redundant sort has been removed and in the `user-accounts` has been deleted default selected vo, so now the default vo is managed by `vo-search-select-component` for this component.